samcli/commands/local/lib/route_collector.py

```diff
@@ -0,0 +1,59 @@
+"""
+Class to store the Route list in the SAM Template. This class helps store both implicit and explicit
+routes in a standardized format
+"""
+
+import logging
+from collections import defaultdict
+
+LOG = logging.getLogger(__name__)
+
+
+class RouteCollector(object):
+
+    def __init__(self):
+        # Route properties stored per resource.
+        self.by_resource = defaultdict(list)
```

Comment:
It is not clear what `by_resource` means here. I know this is what it was before, but this might be a good time to find a new name.

Also, since this is an iterator, I think this should be a private variable to the class. Unless I am missing a use case for directly accessing this var.